mne/stats/cluster_level.py

```diff
@@ -1114,6 +1115,16 @@ def permutation_cluster_test(
     H0 : array, shape (n_permutations,)
         Max cluster level stats observed under permutation.
 
+    Notes
+    -----
+    For computing a ``threshold`` based on a p-value, use the conversion
+    from :meth:`scipy.stats.f.ppf`::
+
+        pval = 0.001  # arbitrary
+        dfn = n_conditions - 1  # degrees of freedom numerator
+        dfd = n_observations - n_conditions  # degrees of freedom denominator
+        thresh = scipy.stats.f.ppf(1 - pval, dfn=dfn, dfd=dfd)
```

Comment:
Depending on the test (anova vs t test) and whether the test is paird this will look slightly different (for rm anova n_subjects is `data[0].shape[0]`, while for independent it is `([dt.shape[0] for dt in data]).sum()`). So I'm wondering whether such info should be here in the notes or in a separate example linked from the docs? In a separate example we could explain all these details. BTW: the permutations should be done differently depending on whether test is paired or not, but I'm not sure how its done in mne now.

Reply:
that's true, but the way these functions are currently set up is that the `_1samp` functions do t-tests and the others do f-tests by default -- see also the return values, and see the "threshold" docstrings for the functions:

- https://github.com/mne-tools/mne-python/blob/965521831b5ca8675becc0cc9716db0cfc8ad946/mne/utils/docs.py#L3043-L3058

We currently have 4 cluster permutation functions, and they cover 2 factors with 2 levels each:

- factor 1 "adjacency": spatio-temporal, or "whatever you want"
- factor 2 "tests": 1-sample t-test, or "whatever you want"

with a proper API (see our roadmap), this would ideally be a single function where one can easily specify parameters. But currently it's designed as if the factor 2 "whatever you want" level is targeted towards F tests, although a user could probably also specify other `stat_funs`.

This is of course not ideal and should be fixed in future PRs -- but I think that's a larger endeavour.

Does somebody have a proposal what concrete steps we could do as part of this PR (with the scope: improve docs without changing the mechanics ... and don't let the perfect be the enemy of the "better" :sweat_smile: )

Reply:
ok, I agree with the perfect being the enemy of good/better :)
I have an API that works quite well for me (at least for various group comparisons, for regression I have a separate cluster_based_regression): https://github.com/mmagnuski/sarna/blob/master/sarna/cluster.py#L523 
and as promised I will try helping to move mne in similar direction